cudamapper/src/overlapper.cpp

```diff
@@ -55,6 +59,34 @@ bool overlaps_mergable(const claraparabricks::genomeworks::cudamapper::Overlap o
     bool gap_ratio_ok = (gap_ratio > 0.8) || ((query_gap < 500) && (target_gap < 500)); //TODO make these user-configurable?
     return gap_ratio_ok;
 }
+
+// Reverse complement lookup table
+static char complement_array[26] = {
+    84, 66, 71, 68, 69,
+    70, 67, 72, 73, 74,
+    75, 76, 77, 78, 79,
+    80, 81, 82, 83, 65,
+    85, 86, 87, 88, 89, 90};
+
+void reverse_complement(std::string& s, const std::size_t len)
+{
+    for (std::size_t i = 0; i < len / 2; ++i)
+    {
+        char tmp       = s[i];
+        s[i]           = static_cast<char>(complement_array[static_cast<int>(s[len - 1 - i] - 65)]);
+        s[len - 1 - i] = static_cast<char>(complement_array[static_cast<int>(tmp) - 65]);
+    }
+}
+std::string string_slice(const std::string& s, const std::size_t start, const std::size_t end)
+{
+    return s.substr(start, end - start);
+}
+
+claraparabricks::genomeworks::cga_string_view_t string_view_slice(const claraparabricks::genomeworks::cga_string_view_t& s, const std::size_t start, const std::size_t end)
```

Comment:
Does this actually has to be a separate function? It's probably a matter of taste...

Reply:
I agree this is a matter of taste, but I like having it as a function like so. Eventually we may want to refactor this to handle overlaps on the "-" strand without reverse complementing the whole string, and having the slice function modularized facilitates that possible change.

Reply:
Ok, I agree we can leave it as it is.